samples/01_contraction/simple_scale_contraction_doubleComplex_cpu.cpp

```diff
@@ -0,0 +1,232 @@
+    int nmodeB = modeB.size();
+    int nmodeD = modeD.size();
+
+    std::unordered_map<int, int64_t> extent;
+    std::unordered_map<int, int64_t> stride;
+    
+    extent['m'] = 5;
+    extent['n'] = 6;
+    extent['u'] = 3;
+    extent['v'] = 4;
+    extent['h'] = 3;
+    extent['k'] = 4;
+
+    stride['m'] = 72;
+    stride['n'] = 12;
+    stride['u'] = 4;
+    stride['v'] = 1;
+    stride['h'] = 4;
+    stride['k'] = 1;
+
+    std::vector<size_t> d_ms_ns_lengths;
+    std::vector<size_t> d_ms_ns_strides;
+    for(auto mode : modeD)
+    {
+        d_ms_ns_lengths.push_back(extent[mode]);
+        d_ms_ns_strides.push_back(stride[mode]);
+    }
+
+    std::vector<size_t> a_ms_ks_lengths;
+    std::vector<size_t> a_ms_ks_strides;
+    for(auto mode : modeA)
+    {
+        a_ms_ks_lengths.push_back(extent[mode]);
+        a_ms_ks_strides.push_back(stride[mode]);
+    }
+
+    std::vector<size_t> b_ns_ks_lengths;
+    std::vector<size_t> b_ns_ks_strides;
+    for(auto mode : modeB)
+    {
+        b_ns_ks_lengths.push_back(extent[mode]);
+        b_ns_ks_strides.push_back(stride[mode]);
+    }
+
+    /**********************
+   * Allocating data
+   **********************/
+    std::cout << "Initializing host data..." << std::endl;
+
+    size_t elementsA = std::accumulate(
+        a_ms_ks_lengths.begin(), a_ms_ks_lengths.end(), size_t{1}, std::multiplies<size_t>());
+    size_t elementsB = std::accumulate(
+        b_ns_ks_lengths.begin(), b_ns_ks_lengths.end(), size_t{1}, std::multiplies<size_t>());
+    size_t elementsD = std::accumulate(
+        d_ms_ns_lengths.begin(), d_ms_ns_lengths.end(), size_t{1}, std::multiplies<size_t>());
+
+    size_t sizeA = sizeof(ADataType) * elementsA;
+    size_t sizeB = sizeof(BDataType) * elementsB;
+    size_t sizeD = sizeof(DDataType) * elementsD;
+
+    ADataType* A = nullptr;
+    BDataType* B = nullptr;
+    DDataType* D = nullptr;
+    CHECK_HIP_ERROR(hipHostMalloc((void**)&A, sizeA));
+    CHECK_HIP_ERROR(hipHostMalloc((void**)&B, sizeB));
+    CHECK_HIP_ERROR(hipHostMalloc((void**)&D, sizeD));
+
+    /*******************
+   * Initialize data
+   *******************/
+    for(int64_t i = 0; i < elementsA; i++)
+    {
+        A[i] = make_hipDoubleComplex(((double(std::rand())) / double(RAND_MAX) - 0.5) * 100,
+                                     ((double(std::rand())) / double(RAND_MAX) - 0.5) * 100);
+    }
+
+    for(int64_t i = 0; i < elementsB; i++)
+    {
+        B[i] = make_hipDoubleComplex(((double(std::rand())) / double(RAND_MAX) - 0.5) * 100,
+                                     ((double(std::rand())) / double(RAND_MAX) - 0.5) * 100);
+    }
+
+    for(int64_t i = 0; i < elementsD; i++)
+    {
+        D[i] = make_hipDoubleComplex((double)std::numeric_limits<double>::signaling_NaN(),
+                                     (double)std::numeric_limits<double>::signaling_NaN());
+    }
+
+    CHECK_HIPTENSOR_ERROR(hiptensorContractionReference((void*)&alpha,
```

Comment:
Just want to confirm that this will be replaced with CK implementation later?

Reply:
yes right, this sample will be a device version like other samples for f32 , f64. Right now, its only a cpu version for complex float and double.